ckanext/datastore/backend/postgres.py

```diff
@@ -1217,16 +1218,21 @@ def upsert_data(context: Context, data_dict: dict[str, Any]):
             rows.append(row)
 
         sql_string = '''INSERT INTO {res_id} ({columns})
-            VALUES ({values});'''.format(
+            VALUES ({values}) {return_statement};'''.format(
             res_id=identifier(data_dict['resource_id']),
             columns=sql_columns,
+            return_statement='RETURNING {return_columns}'.format(
+                return_columns=return_columns) if data_dict.get(
+                    'include_records') else '',
```

Comment:
The validation will guarantee this value exists in data_dict so it's more defensive to use `[]`.

Would be happier if `include_records` was a normal parameter, but I guess this is part of a published interface so it's harder to change?

```suggestion
                return_columns=return_columns) if data_dict[
                    'include_records'] else '',
```